index.js

```diff
@@ -843,7 +830,7 @@ module.exports = function (log, indexesPath) {
   function matchAgainstPrefix(op, prefixIndex, cb) {
     const target = op.data.value
     const targetPrefix = target
-      ? safeReadUint32(target, op.data.prefixOffset)
+      ? safeReadUint32(bipf.slice(target, 0), op.data.prefixOffset)
```

Comment:
What's the story behind this?

Reply:
for prefix we must match the actual buffer and not the type + buffer. bipf.slice will give us the buffer so that it works like before where target was the buffer.

Reply:
Oh :+1: , TIL how bipf.slice actually works